shared/src/test/diff/mlscript/ExprProb.mls

```diff
@@ -138,25 +160,49 @@ eval1 done e1
 pretty1 done e1
 prettier1 done (eval1 done) e1
 //│ e1: Add['E .. 'E | (Add['E0 .. 'E0 | Lit & {val: 2} | Lit & {val: 3}] with {lhs: Lit & {val: 2}, rhs: Lit & {val: 3}}) | Lit & {val: 1}] with {lhs: Lit & {val: 1}, rhs: Add['E0 .. 'E0 | Lit & {val: 2} | Lit & {val: 3}] with {lhs: Lit & {val: 2}, rhs: Lit & {val: 3}}}
+//│   = Add {
+//│       lhs: Lit { val: 1 },
+//│       rhs: Add { lhs: Lit { val: 2 }, rhs: Lit { val: 3 } }
+//│     }
 //│ res: int
+//│    = 6
 //│ res: string
+//│    = Uncaught ReferenceError: concat is not defined
+//│          at REPL23:1:112
 //│ res: string
+//│    = Uncaught ReferenceError: intToString is not defined
+//│          at REPL23:1:69
+//│          at REPL24:1:235
 
 e1 = add (lit 1) (add (lit 2) (lit 3))
 eval1 done e1
 pretty1 done e1
 prettier1 done (eval1 done) e1
 //│ e1: Add['E .. 'E | (Add['E0 .. 'E0 | Lit & {val: 2} | Lit & {val: 3}] with {lhs: Lit & {val: 2}, rhs: Lit & {val: 3}}) | Lit & {val: 1}] with {lhs: Lit & {val: 1}, rhs: Add['E0 .. 'E0 | Lit & {val: 2} | Lit & {val: 3}] with {lhs: Lit & {val: 2}, rhs: Lit & {val: 3}}}
+//│   = '[object Object][object Object][object Object]'
```

Comment:
Whoops... some objects were concatenated?

Reply:
In the era without binary expressions, I hard coded the translation from `add x y` to `x + y`. So the definition here does not work. https://github.com/hkust-taco/mlscript/blob/08d53d1d13507303a8fbfaa9e0edfb1d700c2634/shared/src/test/diff/mlscript/ExprProb.mls#L13